gnsi/acltrie/trie_test.go

```diff
@@ -0,0 +1,320 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package acltrie
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/openconfig/gnmi/errdiff"
+	"github.com/openconfig/ygot/ygot"
+
+	gpb "github.com/openconfig/gnmi/proto/gnmi"
+	pathzpb "github.com/openconfig/gnsi/pathz"
+)
+
+func mustPath(s string) *gpb.Path {
+	p, err := ygot.StringToStructuredPath(s)
+	if err != nil {
+		panic(fmt.Sprintf("cannot parse subscription path %s, %v", s, err))
+	}
+	return p
+}
+
+func TestInsert(t *testing.T) {
```

Comment:
I think a slightly better test would be to check the state of the whole trie. This only checks for errors but doesn't check whether the tree was constructed correctly.

Reply:
if we do that the test become very implementation specific. i'll implement Get if the next PR which will get retrieving policies

Reply:
Sounds good.